examples/semantic/src/main/scala/417c.scala

```diff
@@ -0,0 +1,8 @@
+package ticket417c
+
+trait T {
+  class C
+}
+object T extends T {
+  def foo(): C = ???
```

Comment:
By specifying `this`, we don't hit the `prefix(id)` method since the `TptOps#tpe` method will use `prefix(qual, id)` on seeing a `TptSelect` and give it a `ThisType` prefix.